app/components/palette.hbs

```diff
@@ -0,0 +1,16 @@
+<div class="palette">
+    Selected Color
+    <Swatch
+        @shade={{this.selectedColor}}
+        @size="m"
+        data-test-swatch-selected={{this.selectedColor}}
+     />
+    {{#each this.colorOptions as |option|}}
+        <Swatch
+            @shade={{option}}
+            @size="s"
+            data-test-swatch={{option}}
+            onclick={{action this.updateSelectedColor option}}
```

Comment:
I like the approach of adding the click handler to the component _from the outside_! (especially since it doesn't always support being clicked on) — however, this is currently relying on the "old" semantics of an ember component. (I believe this will stop working if you make the `Swatch` component have `tagName = ''`). 

Instead, you'd want to do three things: 

1. Make the `Swatch` component tagless (with `tagName` = ''`)
2. Add the `...attributes` utility to the Swatch template (which forwards attributes _and_ Modifiers!) 
3. Use the `{{on "click" (fn this.updateSelectedColor option)}}` modifier here! (I switch `action` to `fn` here, too as we should not need to use the `action` helper anywhere we're using Class-based components)